Tests/Functional/Controller/AbstractControllerTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Kitodo\Dlf\Tests\Functional\Controller;
+
+use Kitodo\Dlf\Common\Solr\Solr;
+use Kitodo\Dlf\Controller\AbstractController;
+use Kitodo\Dlf\Domain\Model\SolrCore;
+use Kitodo\Dlf\Domain\Repository\DocumentRepository;
+use Kitodo\Dlf\Domain\Repository\SolrCoreRepository;
+use Kitodo\Dlf\Tests\Functional\FunctionalTestCase;
+use TYPO3\CMS\Extbase\Mvc\Request;
+use TYPO3\CMS\Extbase\Mvc\View\GenericViewResolver;
+use TYPO3\CMS\Extbase\Persistence\Generic\PersistenceManager;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+
+
+abstract class AbstractControllerTest extends FunctionalTestCase
+{
+    public $currentSolrUid = 1;
+
+    public $currentCoreName = '';
+
+    protected function setUpData($databaseFixtures): void
+    {
+        foreach ($databaseFixtures as $filePath) {
+            $this->importCSVDataSet($filePath);
+        }
+        $this->persistenceManager = $this->objectManager->get(PersistenceManager::class);
```

Comment:
This seems to be the problem why the tests fail. `$this->objectManager->get()` throws the error `Call to a member function get() on null`, so `$this->objectManager` isn't initialized.

Reply:
This fails because as of version 12 the GeneralUtilityManager is used to get an instance of a class. However, if I change this, all tests fail. The response only contains an empty string. This must be related to the changes to version 12.
`$this->persistenceManager = GeneralUtility::makeInstance(PersistenceManager::class);`

Reply:
Yes, this fixed the issue. Now the tests are running for both, v11 and v12. Unfortunately, some of the tests fail...